### PR TITLE
refactor(parser): add hint of how tuples are created

### PR DESCRIPTION
**Pull request by vit0rr**
Closes https://github.com/gleam-lang/gleam/issues/3836

### CODE REVIEW

File: compiler-core/src/parse/error.rs [hunk 1]
```diff
@@ -65,7 +65,10 @@ impl ParseError {
             ),
             ParseErrorType::ExprLparStart => (
                 "This parenthesis cannot be understood here",
-                vec!["Hint: To group expressions in gleam use \"{\" and \"}\".".into()],
+                vec![
+                    "Hint: To group expressions in gleam use \"{\" and \"}\".".into(),
+                    "Hint: Tuples are created with `#(` and `)`.".into(),
```
Comment (GearsDatapacks) [inline compiler-core/src/parse/error.rs:70]: Instead of putting "hint" twice, I think it would be better to use one sentence.

Comment (vit0rr) [reply compiler-core/src/parse/error.rs:70]: Agree.
Something like this?

```rust
ParseErrorType::ExprLparStart => (
    "This parenthesis cannot be understood here",
    vec![
        "Hint: To group expressions in Gleam, use \"{\" and \"}\"; tuples are created with `#(` and `)`.".into(),
    ]
),
```

Comment (GearsDatapacks) [reply compiler-core/src/parse/error.rs:70]: Yeah, or just "and"

**Review by lpil — APPROVED** [review]
Thank you!

**Pull request merged**